src/Aggregate/AggregateRepository.php

```diff
@@ -142,20 +144,30 @@ public function addAggregateRoot($eventSourcedAggregateRoot)
             );
         }
 
-        $aggregateId = $this->aggregateTranslator->extractAggregateId($eventSourcedAggregateRoot);
+        $domainEvents = $this->aggregateTranslator->extractPendingStreamEvents($eventSourcedAggregateRoot);
 
-        $domainEvents = new ArrayIterator(
-            $this->aggregateTranslator->extractPendingStreamEvents($eventSourcedAggregateRoot)
+        //We make a copy of the aggregate root to be able to extract the aggregate id
+        //without the need to apply the pending domain events to the real aggregate instance
+        //(this is done in the method applyPendingStreamEvents after EventStore.commit)
+        $aggregateRootCopy = $this->aggregateTranslator->reconstituteAggregateFromHistory(
+            $this->aggregateType,
+            new ArrayIterator($domainEvents)
```

Comment:
good point, but at this time the history should only contain 1 event ;) If the aggregate recorded more than one event the others are maybe required to set the aggregate id correctly (really strange edge-case IMO, but we can't know it)